hapi/hedera-protobufs/services/basic_types.proto

```diff
@@ -1773,3 +1778,62 @@ message PendingAirdropValue {
      */
     uint64 amount = 1;
 }
+
+/**
+ * An approved allowance of hbar transfers for a spender.<br/>
+ * This message SHALL be used to record the allowance of hbars that an account has approved
+ * for sending messages to a given topic.
+ */
+message ConsensusCryptoFeeScheduleAllowance {
+    /**
+     * The account ID of the hbar owner (ie. the grantor of the allowance).
+     */
+    AccountID owner = 1;
+
+    /**
+     * The topic ID enabled to spend fees from the hbar allowance.
+     */
+    TopicID topicId = 2;
+
+    /**
+     * The maximum amount of the spender's allowance in tinybars.
+     */
+    uint64 amount = 3;
+
+    /**
+     * The maximum amount of the spender's token allowance per message.
+     */
+    uint64 amount_per_message = 4;
+}
+
+/**
+ * An approved allowance of hbar transfers for a spender.<br/>
+ * This message SHALL be used to record the allowance of fungible tokens
+ * that an account has approved for sending messages to a given topic.
+ */
+message ConsensusTokenFeeScheduleAllowance {
```

Comment:
Is it intentional to avoid ambiguity to have `ConsensusTokenFeeScheduleAllowance` and `ConsensusCryptoFeeScheduleAllowance`? We could do similar checks that if `token_id` is not set we can assume as crypto allowance and just have `ConsensusTokenFeeScheduleAllowance`

Reply:
I guess its easier to understand, compared to if we have a "hidden" logic that checks the `token_id` and also we are following the pattern from CryptoAllowances

Reply:
The logic won't be hidden, it would be _specified_ here in the protobuf, and part of the HIP.
I would think that having just one ConsensusFeeScheduleAllowance message would be clearer and easier for a client developer to use.

Reply:
@Neeharika-Sompalli @jsync-swirlds can we remove the `owner` field and use the payer of the transaction instead?

Reply:
Probably not.  There are too many use cases (e.g. smart contracts) where the payer submitting a transaction might be (with authorization) creating/updating an allowance for a different entity.

Reply:
I agree; combining these two messages makes sense. However, we should double-check with the author of this HIP. Maybe there is a good reason to keep them separated.

In general in our API, the payer of a transaction is not always the one whose data is modified. This transaction type is not different.

Reply:
The idea was to mimic the current grant behavior where the lists are separated.

There's no strong requirement for this, but I still prefer to use the existing concepts and separate fields because they use different data structures.

I think it's also more future-proof (i.e. if we want to change something on the token side only, or vice versa) and less confusing: let's say we use a single structure and tokenid=0 represents HBAR, this would not be the case with cryptoallowance, and this would lead to confusion.

Simplifications can be done through the SDK anyway, so developers should not be affected even if we use two separate lists.

Reply:
Keeping the existing concept to be consistent with crypto allowances.